c7n/resources/glue.py

```diff
@@ -516,3 +516,37 @@ class GlueCatalogEncryptionFilter(GlueCatalogEncryptionEnabled):
               SseAwsKmsKeyId: alias/aws/glue
 
     """
+
+
+@GlueDataCatalog.filter_registry.register('cross-account')
+class GlueCatalogCrossAccount(CrossAccountAccessFilter):
+    """Filter glue catalog if its has cross account permissions
+
+    :example:
+
+    .. code-block:: yaml
+
+      policies:
+        - name: catalog-cross-account
+          resource: aws.glue-catalog
+          filters:
+            - type: cross-account
+
+    """
+    permissions = ('glue:GetResourcePolicy',)
+    policy_annotation = "c7n:AccessPolicy"
+
+    def get_resource_policy(self, r):
+        client = local_session(self.manager.session_factory).client('glue')
+        if self.policy_annotation in r:
+            return r[self.policy_annotation]
+        try:
+            policy = client.get_resource_policy().get('PolicyInJson')
+        except client.exceptions.EntityNotFoundException:
+            policy = {}
+        r[self.policy_annotation] = policy
+        return policy
+
+    def __call__(self, r):
+        if self.get_resource_policy(r):
```

Comment:
this is pretty broken on logic

Reply:
yep, agree with you @kapilt! I should have given it a thought myself. Switching it back to our logic in `CrossAccountAccessFiter`. I think that'll make it a lot better. Thanks